src/jdk.management/unix/classes/com/sun/management/internal/OperatingSystemImpl.java

```diff
@@ -274,8 +274,11 @@ public double getProcessCpuLoad() {
     }
 
     private boolean isCpuSetSameAsHostCpuSet() {
-        if (containerMetrics != null && containerMetrics.getCpuSetCpus() != null) {
-            return containerMetrics.getCpuSetCpus().length == getHostOnlineCpuCount0();
+        if (containerMetrics != null) {
+            int[] cpuSetCpus = containerMetrics.getCpuSetCpus();
```

Comment:
Might be worth a comment about the volatility of getCpuSetCpus() to head off someone changing it back.

Reply:
I added a comment explaining both the volatility and the potential cost.